multi-tenancy-observability/README.md

```diff
@@ -0,0 +1,270 @@
+
+We will have to either create a new operator to handle multi-tenancy for all observability tools (`observability-multi-tenancy-operator`) or add this feature into the `logging-operator` and rename it to `observability-operator`.
+
+Atlas' current preference is towards the second option : moving towards having a single, global `observability-operator` which would embed several different controllers, each dedicated to its own observability tool (i.e one controller for logging, another one for multi-tenancy, ...). This approach has the benefit of limiting the number of different operators - and thus related repos - which would not only ease the development process but also make troubleshooting a lot whole easier as one would have only 1 operator to check.
+
+We are open to suggestions on this topic.
+
+### Open questions
+
+#### Teleport issue
+
+We are currently facing an issue with `teleport` authentication:
+[Teleport JWT issue](https://github.com/giantswarm/giantswarm/issues/29719)
+
+We are authenticated on Grafana through a JWT token provided by teleport.
+But that token is not forwarded to the datasource as this is not using the Grafana OAuth plugin to authenticate. Due to this, we cannot access the users groups in the grafana-multi-tenant-proxy.
+
+We are considering a number of options that are not ideal at the moment:
+
+- having double datasource: one for customer with OAuth authentication and another dedicated to GiantSwarm people authenticated throw teleport.
+- having double grafana (same idea than above but for the all grafana instance, not just the datasource).
+- share an authentication secret via cookies.
+- other ideas are welcome!
+
+#### Mapping user's groups to tenants
+
+What interface could we offer our customers to enable them to define the mapping between their groups and their tenants:
+
+- Custom resources like TenantCR, GroupCR
+- UI in happa
+- other ideas ?
+
+## Write path
+
+### Current architecture for the write path
+
+On the write path, we currently use promtail to collect container and machine logs and grafana agent to collect kubernetes events.
+
+We currently need those 2 tools because:
+1. Promtail cannot get kubernetes events
+2. Grafana Agent is going through quite a big rewrite in it's configuration and all features we needed were not supported when we deployed it
+
+Here is what the flow looks like:
+1. The logging-operator creates a `password` for the cluster's logging agents to be able to send logs to Loki and configures them (actual configuration of what to scrape and so on) via an extra-config.
+2. The logging-operator adds the new cluster_id/password pair to the proxy configuration
+3. The logging agents collect and send logs to the loki-multi-tenant-proxy which validates the basic authentication sent in the http header and transform it into a tenant information (X-Scope_OrgID)
```

Comment:
What in a future case where customer can define multiple tenants for write path, is that possible? does it require multiple instances of promtail and/or grafana agent?

Reply:
From what we discussed before IIRC we can have multiple tenants , but it would mean some data duplication. But @QuentinBisson can explain it better

Reply:
Yes exactly, so the tenants can be set in promtail as a label via what they called a pipeline stage so if they want they can set multiple tenant, but that means the data will be stored and queries twice so we would advise against it :)

This is why the idea of setting a tenant per namespace makes sense :)